From 8e0c780e619423ed1420582d5f335989bd0a16a1 Mon Sep 17 00:00:00 2001
From: Prabhakar Kushwaha <prabhakar.kushwaha@nxp.com>
Date: Thu, 8 Mar 2018 13:49:05 +0530
Subject: [PATCH 031/107] powerpc: configs: Set initrd_high environment varible

In commit 20e072f, it force to check "bootm_low" and "bootm_size" if
"initrd_high" is missing. It will cause Linux can't boot up with ramdisk
size > 100MB on PPC.

On PPC the initrd does not need to be loaded within the boot mapping.
This assumption of what the absence of initrd_high means in commit
20e072f is not universally appropriate.

As the discussion is continuing in the community, this patch is just a
workaround to fix the issue.

Signed-off-by: Alison Wang <alison.wang@nxp.com>
Signed-off-by: Prabhakar Kushwaha <prabhakar.kushwaha@nxp.com>
---
 include/configs/B4860QDS.h     |    1 +
 include/configs/BSC9131RDB.h   |    1 +
 include/configs/BSC9132QDS.h   |    1 +
 include/configs/C29XPCIE.h     |    1 +
 include/configs/MPC8536DS.h    |    1 +
 include/configs/MPC8541CDS.h   |    1 +
 include/configs/MPC8544DS.h    |    1 +
 include/configs/MPC8548CDS.h   |    1 +
 include/configs/MPC8555CDS.h   |    1 +
 include/configs/MPC8568MDS.h   |    1 +
 include/configs/MPC8569MDS.h   |    1 +
 include/configs/MPC8572DS.h    |    1 +
 include/configs/P1010RDB.h     |    1 +
 include/configs/P1022DS.h      |    1 +
 include/configs/P1023RDB.h     |    1 +
 include/configs/P2041RDB.h     |    1 +
 include/configs/T102xQDS.h     |    1 +
 include/configs/T102xRDB.h     |    1 +
 include/configs/T1040QDS.h     |    1 +
 include/configs/T104xRDB.h     |    1 +
 include/configs/T208xQDS.h     |    1 +
 include/configs/T208xRDB.h     |    1 +
 include/configs/T4240QDS.h     |    1 +
 include/configs/T4240RDB.h     |    1 +
 include/configs/corenet_ds.h   |    1 +
 include/configs/cyrus.h        |    1 +
 include/configs/p1_p2_rdb_pc.h |    1 +
 include/configs/p1_twr.h       |    1 +
 28 files changed, 28 insertions(+), 0 deletions(-)

diff --git a/include/configs/B4860QDS.h b/include/configs/B4860QDS.h
index d28e003..f45d493 100644
--- a/include/configs/B4860QDS.h
+++ b/include/configs/B4860QDS.h
@@ -732,6 +732,7 @@ unsigned long get_board_ddr_clk(void);
 #endif
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	HWCONFIG						\
 	"usb1:dr_mode=host,phy_type=" __stringify(__USB_PHY_TYPE) "\0"\
 	"netdev=eth0\0"						\
diff --git a/include/configs/BSC9131RDB.h b/include/configs/BSC9131RDB.h
index 56e6e14..c0eeb31 100644
--- a/include/configs/BSC9131RDB.h
+++ b/include/configs/BSC9131RDB.h
@@ -332,6 +332,7 @@ extern unsigned long get_sdram_size(void);
 #define CONFIG_UBOOTPATH	"u-boot.bin" /* U-Boot image on TFTP server */
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"netdev=eth0\0"						\
 	"uboot=" CONFIG_UBOOTPATH "\0"				\
 	"loadaddr=1000000\0"			\
diff --git a/include/configs/BSC9132QDS.h b/include/configs/BSC9132QDS.h
index 96a92f1..eb578e8 100644
--- a/include/configs/BSC9132QDS.h
+++ b/include/configs/BSC9132QDS.h
@@ -545,6 +545,7 @@ combinations. this should be removed later
 #endif
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"netdev=eth0\0"						\
 	"uboot=" CONFIG_UBOOTPATH "\0"				\
 	"loadaddr=1000000\0"			\
diff --git a/include/configs/C29XPCIE.h b/include/configs/C29XPCIE.h
index 125a26b..a86401e 100644
--- a/include/configs/C29XPCIE.h
+++ b/include/configs/C29XPCIE.h
@@ -449,6 +449,7 @@
 #define CONFIG_DEF_HWCONFIG	fsl_ddr:ecc=on
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"hwconfig=" __stringify(CONFIG_DEF_HWCONFIG)  "\0"	\
 	"netdev=eth0\0"						\
 	"uboot=" __stringify(CONFIG_UBOOTPATH) "\0"		\
diff --git a/include/configs/MPC8536DS.h b/include/configs/MPC8536DS.h
index 9f0039c..52c3950 100644
--- a/include/configs/MPC8536DS.h
+++ b/include/configs/MPC8536DS.h
@@ -630,6 +630,7 @@
 #define CONFIG_LOADADDR		1000000
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+"initrd_high=0xffffffff\0"					\
 "netdev=eth0\0"						\
 "uboot=" __stringify(CONFIG_UBOOTPATH) "\0"			\
 "tftpflash=tftpboot $loadaddr $uboot; "			\
diff --git a/include/configs/MPC8541CDS.h b/include/configs/MPC8541CDS.h
index 238dd67..3d800f3 100644
--- a/include/configs/MPC8541CDS.h
+++ b/include/configs/MPC8541CDS.h
@@ -376,6 +376,7 @@ extern unsigned long get_clock_freq(void);
 #define CONFIG_LOADADDR  200000   /*default location for tftp and bootm*/
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				        \
+   "initrd_high=0xffffffff\0"						\
    "netdev=eth0\0"                                                      \
    "consoledev=ttyS1\0"                                                 \
    "ramdiskaddr=600000\0"                                               \
diff --git a/include/configs/MPC8544DS.h b/include/configs/MPC8544DS.h
index 64e0aa1..c74a324 100644
--- a/include/configs/MPC8544DS.h
+++ b/include/configs/MPC8544DS.h
@@ -384,6 +384,7 @@ extern unsigned long get_board_sys_clk(unsigned long dummy);
 #define CONFIG_LOADADDR	1000000	/*default location for tftp and bootm*/
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+"initrd_high=0xffffffff\0"					\
 "netdev=eth0\0"						\
 "uboot=" __stringify(CONFIG_UBOOTPATH) "\0"			\
 "tftpflash=tftpboot $loadaddr $uboot; "			\
diff --git a/include/configs/MPC8548CDS.h b/include/configs/MPC8548CDS.h
index cdd70a1..c308f90 100644
--- a/include/configs/MPC8548CDS.h
+++ b/include/configs/MPC8548CDS.h
@@ -492,6 +492,7 @@ extern unsigned long get_clock_freq(void);
 #define CONFIG_LOADADDR	1000000	/*default location for tftp and bootm*/
 
 #define	CONFIG_EXTRA_ENV_SETTINGS		\
+	"initrd_high=0xffffffff\0"		\
 	"hwconfig=fsl_ddr:ecc=off\0"		\
 	"netdev=eth0\0"				\
 	"uboot=" __stringify(CONFIG_UBOOTPATH) "\0"	\
diff --git a/include/configs/MPC8555CDS.h b/include/configs/MPC8555CDS.h
index bcef1e7..ecad9c7 100644
--- a/include/configs/MPC8555CDS.h
+++ b/include/configs/MPC8555CDS.h
@@ -372,6 +372,7 @@ extern unsigned long get_clock_freq(void);
 #define CONFIG_LOADADDR  200000   /*default location for tftp and bootm*/
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				        \
+   "initrd_high=0xffffffff\0"						\
    "netdev=eth0\0"                                                      \
    "consoledev=ttyS1\0"                                                 \
    "ramdiskaddr=600000\0"                                               \
diff --git a/include/configs/MPC8568MDS.h b/include/configs/MPC8568MDS.h
index 6d9a964..0fa3cfa 100644
--- a/include/configs/MPC8568MDS.h
+++ b/include/configs/MPC8568MDS.h
@@ -390,6 +390,7 @@ extern unsigned long get_clock_freq(void);
 #define CONFIG_LOADADDR  200000   /*default location for tftp and bootm*/
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				        \
+   "initrd_high=0xffffffff\0"						\
    "netdev=eth0\0"                                                      \
    "consoledev=ttyS0\0"                                                 \
    "ramdiskaddr=600000\0"                                               \
diff --git a/include/configs/MPC8569MDS.h b/include/configs/MPC8569MDS.h
index ae92c37..9239b43 100644
--- a/include/configs/MPC8569MDS.h
+++ b/include/configs/MPC8569MDS.h
@@ -483,6 +483,7 @@ extern unsigned long get_clock_freq(void);
 #define CONFIG_LOADADDR  200000   /*default location for tftp and bootm*/
 
 #define	CONFIG_EXTRA_ENV_SETTINGS					\
+	"initrd_high=0xffffffff\0"					\
 	"netdev=eth0\0"							\
 	"consoledev=ttyS0\0"						\
 	"ramdiskaddr=600000\0"						\
diff --git a/include/configs/MPC8572DS.h b/include/configs/MPC8572DS.h
index 303922a..ee8dbe2 100644
--- a/include/configs/MPC8572DS.h
+++ b/include/configs/MPC8572DS.h
@@ -586,6 +586,7 @@
 #define CONFIG_LOADADDR		1000000
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+"initrd_high=0xffffffff\0"					\
 "hwconfig=fsl_ddr:ctlr_intlv=bank,bank_intlv=cs0_cs1,ecc=off\0"		\
 "netdev=eth0\0"						\
 "uboot=" __stringify(CONFIG_UBOOTPATH) "\0"				\
diff --git a/include/configs/P1010RDB.h b/include/configs/P1010RDB.h
index 344d99d..55042c1 100644
--- a/include/configs/P1010RDB.h
+++ b/include/configs/P1010RDB.h
@@ -732,6 +732,7 @@ extern unsigned long get_sdram_size(void);
 #define CONFIG_LOADADDR		1000000
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"hwconfig=" __stringify(CONFIG_DEF_HWCONFIG)  "\0"	\
 	"netdev=eth0\0"						\
 	"uboot=" __stringify(CONFIG_UBOOTPATH) "\0"		\
diff --git a/include/configs/P1022DS.h b/include/configs/P1022DS.h
index 62165df..7d0c51c 100644
--- a/include/configs/P1022DS.h
+++ b/include/configs/P1022DS.h
@@ -599,6 +599,7 @@
 #define CONFIG_LOADADDR		1000000
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"netdev=eth0\0"						\
 	"uboot=" __stringify(CONFIG_UBOOTPATH) "\0"		\
 	"ubootaddr=" __stringify(CONFIG_SYS_TEXT_BASE) "\0"	\
diff --git a/include/configs/P1023RDB.h b/include/configs/P1023RDB.h
index 8dbfb06..64a1696 100644
--- a/include/configs/P1023RDB.h
+++ b/include/configs/P1023RDB.h
@@ -312,6 +312,7 @@ extern unsigned long get_clock_freq(void);
 #endif
 
 #define CONFIG_EXTRA_ENV_SETTINGS	\
+	"initrd_high=0xffffffff\0"				\
 	"netdev=eth0\0"						\
 	"uboot=" __stringify(CONFIG_UBOOTPATH) "\0"		\
 	"loadaddr=1000000\0"					\
diff --git a/include/configs/P2041RDB.h b/include/configs/P2041RDB.h
index bf9b1c4..3bfc35c 100644
--- a/include/configs/P2041RDB.h
+++ b/include/configs/P2041RDB.h
@@ -608,6 +608,7 @@ unsigned long get_board_sys_clk(unsigned long dummy);
 #define __USB_PHY_TYPE	utmi
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"hwconfig=fsl_ddr:ctlr_intlv=cacheline,"		\
 	"bank_intlv=cs0_cs1\0"					\
 	"netdev=eth0\0"						\
diff --git a/include/configs/T102xQDS.h b/include/configs/T102xQDS.h
index 8001e70..a9b22d0 100644
--- a/include/configs/T102xQDS.h
+++ b/include/configs/T102xQDS.h
@@ -777,6 +777,7 @@ unsigned long get_board_ddr_clk(void);
 #define __USB_PHY_TYPE		utmi
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"hwconfig=fsl_ddr:ctlr_intlv=cacheline,bank_intlv=cs0_cs1;\0"  \
 	"usb1:dr_mode=host,phy_type=" __stringify(__USB_PHY_TYPE) "\0" \
 	"bootargs=root=/dev/ram rw console=ttyS0,115200\0" \
diff --git a/include/configs/T102xRDB.h b/include/configs/T102xRDB.h
index e80ba14..3d3b46a 100644
--- a/include/configs/T102xRDB.h
+++ b/include/configs/T102xRDB.h
@@ -799,6 +799,7 @@ unsigned long get_board_ddr_clk(void);
 #endif
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"hwconfig=fsl_ddr:ctlr_intlv=cacheline,"		\
 	"bank_intlv=" __stringify(BANK_INTLV) "\0"		\
 	"usb1:dr_mode=host,phy_type=" __stringify(__USB_PHY_TYPE) "\0"  \
diff --git a/include/configs/T1040QDS.h b/include/configs/T1040QDS.h
index c0a1bb4..23a0d89 100644
--- a/include/configs/T1040QDS.h
+++ b/include/configs/T1040QDS.h
@@ -660,6 +660,7 @@ unsigned long get_board_ddr_clk(void);
 #define __USB_PHY_TYPE	utmi
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"hwconfig=fsl_ddr:bank_intlv=auto;"			\
 	"usb1:dr_mode=host,phy_type=" __stringify(__USB_PHY_TYPE) "\0"\
 	"netdev=eth0\0"						\
diff --git a/include/configs/T104xRDB.h b/include/configs/T104xRDB.h
index a8f7885..a054b97 100644
--- a/include/configs/T104xRDB.h
+++ b/include/configs/T104xRDB.h
@@ -809,6 +809,7 @@ $(SRCTREE)/board/freescale/t104xrdb/t1042d4_sd_rcw.cfg
 #endif
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"hwconfig=fsl_ddr:bank_intlv=cs0_cs1;"			\
 	"usb1:dr_mode=host,phy_type=" __stringify(__USB_PHY_TYPE) ";"\
 	"usb2:dr_mode=host,phy_type=" __stringify(__USB_PHY_TYPE) "\0"\
diff --git a/include/configs/T208xQDS.h b/include/configs/T208xQDS.h
index 50413eb..a5539f3 100644
--- a/include/configs/T208xQDS.h
+++ b/include/configs/T208xQDS.h
@@ -746,6 +746,7 @@ unsigned long get_board_ddr_clk(void);
 #define __USB_PHY_TYPE		utmi
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"hwconfig=fsl_ddr:"					\
 	"ctlr_intlv=" __stringify(CTRL_INTLV_PREFERED) ","	\
 	"bank_intlv=auto;"					\
diff --git a/include/configs/T208xRDB.h b/include/configs/T208xRDB.h
index ca1916a..48eb1d0 100644
--- a/include/configs/T208xRDB.h
+++ b/include/configs/T208xRDB.h
@@ -693,6 +693,7 @@ unsigned long get_board_ddr_clk(void);
 #define __USB_PHY_TYPE		utmi
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"hwconfig=fsl_ddr:"					\
 	"ctlr_intlv=" __stringify(CTRL_INTLV_PREFERED) ","	\
 	"bank_intlv=auto;"					\
diff --git a/include/configs/T4240QDS.h b/include/configs/T4240QDS.h
index 597ba2e..9316100 100644
--- a/include/configs/T4240QDS.h
+++ b/include/configs/T4240QDS.h
@@ -516,6 +516,7 @@ unsigned long get_board_ddr_clk(void);
 #endif
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"hwconfig=fsl_ddr:"					\
 	"ctlr_intlv=" __stringify(CTRL_INTLV_PREFERED) ","	\
 	"bank_intlv=auto;"					\
diff --git a/include/configs/T4240RDB.h b/include/configs/T4240RDB.h
index 87d65e4..21d100a 100644
--- a/include/configs/T4240RDB.h
+++ b/include/configs/T4240RDB.h
@@ -667,6 +667,7 @@ unsigned long get_board_ddr_clk(void);
 #endif
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"hwconfig=fsl_ddr:"					\
 	"ctlr_intlv=" __stringify(CTRL_INTLV_PREFERED) ","	\
 	"bank_intlv=auto;"					\
diff --git a/include/configs/corenet_ds.h b/include/configs/corenet_ds.h
index 83c2642..b6688b0 100644
--- a/include/configs/corenet_ds.h
+++ b/include/configs/corenet_ds.h
@@ -621,6 +621,7 @@
 #endif
 
 #define	CONFIG_EXTRA_ENV_SETTINGS				\
+	"initrd_high=0xffffffff\0"				\
 	"hwconfig=fsl_ddr:ctlr_intlv=cacheline,"		\
 	"bank_intlv=cs0_cs1;"					\
 	"usb1:dr_mode=host,phy_type=" __stringify(__USB_PHY_TYPE) ";"\
diff --git a/include/configs/cyrus.h b/include/configs/cyrus.h
index dde0dc3..41bb552 100644
--- a/include/configs/cyrus.h
+++ b/include/configs/cyrus.h
@@ -440,6 +440,7 @@
 #define __USB_PHY_TYPE	utmi
 
 #define	CONFIG_EXTRA_ENV_SETTINGS \
+"initrd_high=0xffffffff\0"				\
 "hwconfig=fsl_ddr:ctlr_intlv=cacheline,"		\
 "bank_intlv=cs0_cs1;"					\
 "usb1:dr_mode=host,phy_type=" __stringify(__USB_PHY_TYPE) ";"\
diff --git a/include/configs/p1_p2_rdb_pc.h b/include/configs/p1_p2_rdb_pc.h
index d0f5529..624f9eb 100644
--- a/include/configs/p1_p2_rdb_pc.h
+++ b/include/configs/p1_p2_rdb_pc.h
@@ -841,6 +841,7 @@ i2c mw 18 3 __SW_BOOT_MASK 1; reset
 #endif
 
 #define	CONFIG_EXTRA_ENV_SETTINGS	\
+"initrd_high=0xffffffff\0"		\
 "netdev=eth0\0"	\
 "uboot=" __stringify(CONFIG_UBOOTPATH) "\0"	\
 "loadaddr=1000000\0"	\
diff --git a/include/configs/p1_twr.h b/include/configs/p1_twr.h
index e02c31d..5a2826b 100644
--- a/include/configs/p1_twr.h
+++ b/include/configs/p1_twr.h
@@ -404,6 +404,7 @@ extern unsigned long get_board_sys_clk(unsigned long dummy);
 #define CONFIG_LOADADDR	1000000
 
 #define	CONFIG_EXTRA_ENV_SETTINGS	\
+"initrd_high=0xffffffff\0"		\
 "netdev=eth0\0"	\
 "uboot=" __stringify(CONFIG_UBOOTPATH) "\0"	\
 "loadaddr=1000000\0"	\
-- 
1.7.1

